exist-core/src/main/java/org/exist/xmldb/XmldbURI.java

```diff
@@ -425,7 +361,7 @@ public URI getURI() {
     }
 
     /**
-     * This returns an xmldb uri. This is the most generic sort of uri - the only fields set in the uri are scheme and schemeSpecificPart
+     * This returns a xmldb uri. This is the most generic sort of uri - the only fields set in the uri are scheme and schemeSpecificPart
```

Comment:
In english, it should still be "an xmldb uri". The "a xmldb uri" does not make sense.

Reply:
Then IntelliJ's spell checker must be at fault.

Reply:
Okay, so let's change it back

Reply:
This case (determining when an "x" is voiced as a consonant or a vowel, for the purpose of selecting the appropriate article "a" vs. "an") is likely to trip up any spell checker, particularly with domain-specific terms like "XMLDB" that may be spelled in a lower case form. See a similar question about X-ray: https://english.stackexchange.com/questions/8181/does-one-use-a-or-an-before-the-word-x-ray. Lots of other good examples in the comments. Since machines may not get it right, we have community review!

Reply:
What about HTTP and HTML?
My feeling says (now backed by @joewiz' explanation)

- an HTML page
- an http-request

correct?

Reply:
It depends to a certain extent what follows "HTTP" or "HTML". I often have to read a phrase out loud (in my head) to decide if it is an "a" or an "an", I'm not sure of the exact rules... but for me personally one of the approaches will seem correct after I sound it out a couple of times.

Reply:
To answer your question:

> an HTML page
> an http-request

I would write both of those as "a", and not "an". Context is important.

Reply:
I remember once discussing this very "HTML" issue with Adam some years back! We were correcting each other's articles (i.e., "a HTML" vs. "an HTML"). And finally (I think?) we had a call and realized why. He pronounces the letter "h" as "haych", whereas I pronounce it "aych." This explains why he suggests "a HTML page" and I would write "an HTML page." So we're getting into regional differences. I'm pretty sure an en-US spell checked would say "an HTML page," but I'm not sure about an en-UK or en-AU, etc. All to say, English is a crazy language!